final class kotlinx.coroutines.JobSupport$AwaitContinuation extends kotlinx.coroutines.CancellableContinuationImpl
{
    private final kotlinx.coroutines.JobSupport job;

    public void <init>(kotlin.coroutines.Continuation, kotlinx.coroutines.JobSupport)
    {
        kotlinx.coroutines.JobSupport$AwaitContinuation r0;
        kotlin.coroutines.Continuation $r1;
        kotlinx.coroutines.JobSupport $r2;

        r0 := @this: kotlinx.coroutines.JobSupport$AwaitContinuation;

        $r1 := @parameter0: kotlin.coroutines.Continuation;

        $r2 := @parameter1: kotlinx.coroutines.JobSupport;

        specialinvoke r0.<kotlinx.coroutines.CancellableContinuationImpl: void <init>(kotlin.coroutines.Continuation,int)>($r1, 1);

        r0.<kotlinx.coroutines.JobSupport$AwaitContinuation: kotlinx.coroutines.JobSupport job> = $r2;

        return;
    }

    public java.lang.Throwable getContinuationCancellationCause(kotlinx.coroutines.Job)
    {
        kotlinx.coroutines.JobSupport$AwaitContinuation r0;
        kotlinx.coroutines.Job $r1;
        kotlinx.coroutines.JobSupport $r2;
        java.lang.Object $r3;
        boolean $z0;
        kotlinx.coroutines.JobSupport$Finishing $r4;
        java.lang.Throwable $r5;
        kotlinx.coroutines.CompletedExceptionally $r6;
        java.util.concurrent.CancellationException $r7;

        r0 := @this: kotlinx.coroutines.JobSupport$AwaitContinuation;

        $r1 := @parameter0: kotlinx.coroutines.Job;

        $r2 = r0.<kotlinx.coroutines.JobSupport$AwaitContinuation: kotlinx.coroutines.JobSupport job>;

        $r3 = virtualinvoke $r2.<kotlinx.coroutines.JobSupport: java.lang.Object getState$kotlinx_coroutines_core()>();

        $z0 = $r3 instanceof kotlinx.coroutines.JobSupport$Finishing;

        if $z0 == 0 goto label2;

        $r4 = (kotlinx.coroutines.JobSupport$Finishing) $r3;

        $r5 = virtualinvoke $r4.<kotlinx.coroutines.JobSupport$Finishing: java.lang.Throwable getRootCause()>();

        if $r5 != null goto label1;

        goto label2;

     label1:
        return $r5;

     label2:
        $z0 = $r3 instanceof kotlinx.coroutines.CompletedExceptionally;

        if $z0 == 0 goto label3;

        $r6 = (kotlinx.coroutines.CompletedExceptionally) $r3;

        $r5 = $r6.<kotlinx.coroutines.CompletedExceptionally: java.lang.Throwable cause>;

        return $r5;

     label3:
        $r7 = interfaceinvoke $r1.<kotlinx.coroutines.Job: java.util.concurrent.CancellationException getCancellationException()>();

        $r5 = (java.lang.Throwable) $r7;

        return $r5;
    }

    protected java.lang.String nameString()
    {
        kotlinx.coroutines.JobSupport$AwaitContinuation r0;

        r0 := @this: kotlinx.coroutines.JobSupport$AwaitContinuation;

        return "AwaitContinuation";
    }
}
